pkg/distributor/distributor.go

```diff
@@ -599,6 +599,13 @@ func (d *Distributor) parseStreamLabels(vContext validationContext, key string,
 //
 // desiredRate is expected to be given in bytes.
 func (d *Distributor) shardCountFor(stream *logproto.Stream, streamSize, desiredRate int, rateStore RateStore) int {
+	if desiredRate <= 0 {
+		if d.cfg.ShardStreams.LoggingEnabled {
+			level.Error(util_log.Logger).Log("msg", "invalid desired rate", "desired_rate", desiredRate)
```

Comment:
It will be once I merge https://github.com/grafana/loki/pull/7311. I added it for now but if you think it is a better idea to merge this without logging the tenant and only add the tenant within 7311/in the future, I can revert it.